install.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+
+cd /tmp/
+
+if [ "$(uname -s)" = "Darwin" ] && [ "$(uname -m)" = "x86_64" ]; then
+    target="x86_64-apple-darwin"
+elif [ "$(uname -s)" = "Linux" ] && [ "$(uname -m)" = "x86_64" ]; then
+    target="x86_64-unknown-linux-musl"
+elif [ "$(uname -s)" = "Linux" ] && [ "$(uname -m)" = "amd64" ]; then
+    target="x86_64-unknown-linux-musl"
+elif [ "$(uname -s)" = "Linux" ] && [ "$(uname -m)" = "arm" ]; then
+    target="arm-unknown-linux-gnueabihf"
+else
+    echo "unsupported OS or architecture"
+    exit 1
+fi
+
+rm -rf xh*
+
+curl -s https://api.github.com/repos/ducaale/xh/releases/latest \
+| grep -wo "https.*$target.tar.gz" \
+| wget -qi -
```

Comment:
This uses both `wget` and `curl`. I think it's better to only use one of them, and ideal to detect which one is available.

Reply:
I wonder if we need to check the existence of `curl` since the script is going to be downloaded using curl.

Reply:
That would be nice, yes. I don't think it's installed by default on all Linux distros.

If we want to also support `wget`, then `wget -nv -O` is roughly equivalent to `curl -sSL -o`. Both support `-` to write to stdout. So you could have something like this:

```sh
if which curl > /dev/null; then
    fetch='curl -sSL -o'
elif which wget > /dev/null; then
    fetch='wget -q -O'
else
    echo "Can't find curl or wget, can't download package"
    exit 1
fi
```

And then later on:

```sh
$fetch - https://api.github.com/repos/ducaale/xh/releases/latest
```
and
```sh
$fetch xh.tar.gz "$url"
```